File: 3461-find-the-minimum-area-to-cover-all-ones-i/find-the-minimum-area-to-cover-all-ones-i.java
class Solution {
    public int minimumArea(int[][] grid) 
    {
        int top = -1, bottom = -1, left = -1, right = -1;
        for(int i = 0; i < grid.length; i++)
        {
            for(int j = 0; j < grid[0].length; j++)
            {
                if(grid[i][j] == 0)
                continue;

                if(top == -1)
                {
                    top = bottom = i;
                    left = right = j;
                }
                else
                {
                    bottom = Math.max(bottom, i);
                    left = Math.min(left, j);
                    right = Math.max(right, j);
                }
            }
        }

        int maxArea = (bottom - top + 1) * (right - left + 1);
        return top == -1 ? 0 : maxArea;
    }
}

File: 3461-find-the-minimum-area-to-cover-all-ones-i/find-the-minimum-area-to-cover-all-ones-i.cpp
class Solution {
public:
    int minimumArea(vector<vector<int>>& grid) 
    {
        int n = grid.size(), m = grid[0].size();
        int top = n, bottom = 0, left = m, right = 0;
        for(int i = 0; i < n; i++)
        {
            for(int j = 0; j < m; j++)
            {
                if(grid[i][j] == 0)
                continue;

                top = min(top, i);
                bottom = max(bottom, i);
                left = min(left, j);
                right = max(right, j);
            }
        }
        return  (bottom - top + 1) * (right - left + 1);
    }
};